lib/cdo/lesson_import_helper.rb

```diff
@@ -396,11 +357,26 @@ def self.find_activity_section_names(markdown)
     markdown.to_enum(:scan, regex).map {Regexp.last_match}
   end
 
-  # No levels, no tips, just markdown
-  def self.create_basic_activity_section(markdown)
+  # We preprocess activity markdown to look for remarks, pullthrough, names, and tips
+  # but we need to look for tips (which may not be at the beginning of a line) and
+  # create pure markdown sections.
+  def self.create_activity_sections_from_markdown(markdown, tip_match_map)
+    sections = []
     stripped_markdown = markdown.strip
-    description = stripped_markdown
-    ActivitySection.new(description: description)
+    tip = find_first_tip_link(stripped_markdown)
```

Comment:
This is the implementation of the new tip logic. It looks for new tips, creating activity sections as it goes.

For [example](https://www.codecurricula.com/csp-21/unit4/9/):
```**Distribute:** Share print copies of [r song-lyrics-functions] or project the activity guide where all can see it.

[slide] discussion!!!discussion-0 **Prompt:**

**Discuss:** Have students silently consider these prompts on their own, then share with a neighbor, and finally move to a full class discussion. 
```

The first tip is at `[slide] discussion!!!discussion-0 ***Prompt:***` so we create an activity section with everything above in on line 370, then create another activity section with the tip on line 373. Then there are no more tips found, so the while loops breaks and we create an activity section with the rest of the content on line 378.

So the descriptions of the 3 activity sections are:
1. `*Distribute:** Share print copies of [r song-lyrics-functions] or project the activity guide where all can see it.`
2. `[slide] **Prompt:**` (the tip syntax is stripped out)
3. `**Discuss:** Have students silently consider these prompts on their own, then share with a neighbor, and finally move to a full class discussion.`